### PR TITLE
Review CIS RHEL8 v3.0.0 Section 4 and Section 5

### PR DESCRIPTION
**Pull request by marcusburghardt**
#### Description:

Review the cis_rhel8.yml control file and update the following sections  in alignment to CIS RHEL 8 v3.0.0:
- Section 4 - Access, Authentication and Authorization
- Section 5 - Logging and Auditing

Also update references in related rules.

#### Rationale:

Keep RHEL 8 profiles updated with CIS RHEL 8 last version.

#### Review Hints:

NOTE: Many requirements were swapped between section 4 and 5. This is the reason this PR includes the review of both sections.

The easiest way is to open the CIS RHEL8 v3.0.0 policy and compare the changes in commits with the respective requirements.

### CODE REVIEW

**Comment by marcusburghardt** [comment]
The conflicts here are basically the same I had to resolve in https://github.com/ComplianceAsCode/content/pull/11455.
I will wait https://github.com/ComplianceAsCode/content/pull/11455 to be merged first then I rebase this PR.

**Comment by marcusburghardt** [comment]
I rebased this PR on top of the latest upstream master branch.

**Comment by marcusburghardt** [comment]
Rebased again.

**Review by vojtapolasek — CHANGES_REQUESTED** [review]
Hello @marcusburghardt  and thank you for this big update. I have only minor comments, it looks mostly good.

File: controls/cis_rhel8.yml [hunk 1]
```diff
@@ -1531,298 +1531,821 @@ controls:
...
     levels:
-      - l2_server
-      - l2_workstation
+      - l1_server
+      - l1_workstation
     status: automated
     rules:
-      - audit_rules_suid_auid_privilege_function
+      - file_at_deny_not_exist
+      - file_groupowner_at_allow
```
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply controls/cis_rhel8.yml:1629]: In combination with the rule to ensure there is no at.deny file I consider enough to satisfy this requirement. But I think it is worth for a note. I will include a note here.

File: controls/cis_rhel8.yml [hunk 2]
```diff
@@ -1531,298 +1531,821 @@ controls:
...
     status: automated
+    notes: |-
+      Introduced in CIS RHEL8 v3.0.0
     rules:
-      - audit_rules_usergroup_modification_group
-      - audit_rules_usergroup_modification_gshadow
-      - audit_rules_usergroup_modification_opasswd
-      - audit_rules_usergroup_modification_passwd
-      - audit_rules_usergroup_modification_shadow
+      - sshd_use_approved_ciphers
```
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply controls/cis_rhel8.yml:1695]: Great

File: controls/cis_rhel8.yml [hunk 3]
```diff
@@ -1531,298 +1531,821 @@ controls:
...
-    rules:
-      - audit_rules_file_deletion_events_rename
-      - audit_rules_file_deletion_events_renameat
-      - audit_rules_file_deletion_events_unlink
-      - audit_rules_file_deletion_events_unlinkat
+      - l1_server
+      - l1_workstation
+    status: pending
+    notes: |-
+      Rule sshd_use_strong_kex needs to be enabled for rhel8.
```
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply controls/cis_rhel8.yml:1749]: For sure. I was not sure about this some days ago. : ) I am updating now.

File: controls/cis_rhel8.yml [hunk 4]
```diff
@@ -1531,298 +1531,821 @@ controls:
...
-      - l2_server
-      - l2_workstation
-    status: automated
-    rules:
-      - audit_rules_execution_setfacl
+      - l1_server
+      - l1_workstation
+    status: pending
+    notes: |-
+      Rule sshd_use_strong_macs needs to be enabled for rhel8.
```
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply controls/cis_rhel8.yml:1786]: done.

File: controls/cis_rhel8.yml [hunk 5]
```diff
@@ -1531,298 +1531,821 @@ controls:
...
+      - var_accounts_passwords_pam_faillock_deny=5
+
+  - id: 4.4.3.1.2
+    title: Ensure password unlock time is configured (Automated)
+    levels:
+      - l1_server
+      - l1_workstation
+    status: automated
+    rules:
+      - accounts_passwords_pam_faillock_unlock_time
```
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply controls/cis_rhel8.yml:2043]: Its a fair point. I will include a note.

File: controls/cis_rhel8.yml [hunk 6]
```diff
@@ -1531,298 +1531,821 @@ controls:
...
-      - l2_workstation
-    status: manual
+      - l1_server
+      - l1_workstation
+    status: automated
+    rules:
+      - accounts_umask_etc_bashrc
+      - accounts_umask_etc_login_defs
+      - accounts_umask_etc_profile
+      - accounts_umask_interactive_users
```
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply controls/cis_rhel8.yml:2345]: Aligned.

File: controls/cis_rhel8.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply controls/cis_rhel8.yml:1632]: Done.

File: controls/cis_rhel8.yml [hunk 7]
```diff
@@ -1531,298 +1531,821 @@ controls:
...
-  - id: 4.1.3.3
-    title: Ensure events that modify the sudo log file are collected (Automated)
+  - id: 4.2.1
+    title: Ensure permissions on /etc/ssh/sshd_config are configured (Automated)
     levels:
-      - l2_server
-      - l2_workstation
+      - l1_server
+      - l1_workstation
     status: automated
```
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply controls/cis_rhel8.yml]: Done

File: controls/cis_rhel8.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply controls/cis_rhel8.yml:1701]: Done

File: controls/cis_rhel8.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply controls/cis_rhel8.yml]: Done

File: controls/cis_rhel8.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply controls/cis_rhel8.yml:2050]: Done

File: controls/cis_rhel8.yml [hunk 8]
```diff
@@ -1981,564 +2495,442 @@ controls:
...
-  - id: 5.2.16
-    title: Ensure SSH MaxAuthTries is set to 4 or less (Automated)
+  - id: 5.2.3.5
+    title: Ensure events that modify the system's network environment are collected (Automated)
     levels:
-      - l1_server
-      - l1_workstation
+      - l2_server
+      - l2_workstation
     status: automated
```
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply controls/cis_rhel8.yml]: Done.

**Comment by marcusburghardt** [comment]
> Hello @marcusburghardt and thank you for this big update. I have only minor comments, it looks mostly good.

Thanks for the review @vojtapolasek . I addressed all the comments in the last commits.

**Comment by vojtapolasek** [comment]
Looks great now @marcusburghardt , just waiting for CI to finish.

**Review by vojtapolasek — APPROVED** [review]

**Comment by vojtapolasek** [comment]
Waiving tests on Rawhide, Rawhide is currently broken. The PR does not touch anything related to Kubernetes so waiving as well.

**Pull request merged**